src/components/VTable.vue

```diff
@@ -0,0 +1,154 @@
+          <img src="../assets/img/loader.gif" alt="">
+        </slot>
+      </div>
+    </div>
+    <table v-if="dataSource.length && columns.length">
+      <thead> <!-- Header block -->
+        <tr>
+          <th
+            v-for="(header, idx) in settings.headers"
+            :key="idx"
+            :class="header._otoptions.class"
+            :style="header._otoptions.style"
+          >
+            <slot
+              :name="header._otoptions.slotName"
+              :header="header"
+            > <!-- Slot for transmiting users data to header's cell -->
+              <span>
+                {{ header.displayName }}
+              </span>
+            </slot>
+          </th>
+        </tr>
+      </thead>
+      <tbody> <!-- Main body block -->
+        <tr
+          v-for="(element, idxLine) in settings.mainBodyItems"
+          :key="idxLine"
+          @click="$emit('rowClick', element)"
+        > <!-- rowClick function emits index of the row on the top -->
+          <td
+            v-for="(header, idx) in settings.headers"
+            :key="idx"
+            :style="header._otoptions.style"
+          >
+            <slot
+              :name="`line-${idxLine}_cell-${idx}`"
+              :bodyCell="element[header.displayValue]"
+            >
+              <span> <!-- Slot for transmiting users data to main body table cell -->
+                {{ element[header.displayValue] }}
+              </span>
+            </slot>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+    <div v-else class="empty-table">
+      <slot name="empty-table"> <!-- Slot will be shown in case no data was provided -->
+        No data provided
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script>
+import { computed, ref } from 'vue'
+
+export default {
+  name: 'main-table',
+  props: {
+    dataSource: { // Here must be trasmited an array with objects for showing header
+      type: Array,
+      default: () => []
+    },
+    columns: { // Here must be trasmited an array with objects for showing main body
+      type: Array,
+      default: () => []
+    },
+    isLoading: { // This prop switch off or switch on loader, Boolean property is expected
+      type: Boolean,
+      default: true
+    }
+  },
+  setup (props) {
+    /* eslint-disable */
+    const settings = computed(() => {
+      return {
+        headers: props.columns
+                  .filter((header) => header.displayValue && header.displayName)
+                  .map((item, idx, array) => {
+                    return {
+                      ...item,
+                      _otoptions: {
+                        // creates class name for header table cell
+                        class: `col_name-${item.displayValue}`,
+                        // creates right line of the table, except for the last column
+                        style: idx !== array.length - 1 ? 'border-right: 1px solid #EEF1F2' : '',
```

Comment:
style should be object
makeHeaderStyle =(array) => ({
       borderRight: .....
})
style: makeHeaderStyle(array)

Reply:
done